extensions/phylum.ts

```diff
@@ -20,6 +20,37 @@ export type ProcessOutput = {
   code: number | null;
 };
 
+export type PolicyEvaluationResponseRaw = {
+  is_failure: boolean;
+  incomplete_packages_count: number;
+  help: string;
+  job_link: string | null;
+  dependencies: EvaluatedDependencies[];
+};
+
+export type EvaluatedDependencies = {
+  purl: string;
+  registry: string;
+  name: string;
+  version: string;
+  rejections: PolicyRejection[];
+};
+
+export type PolicyRejection = {
+  title: string;
+  suppressed: boolean;
+  source: RejectionSource;
+};
+
+export type RejectionSource = {
+  type: string;
+  tag: string | null;
+  domain: string | null;
+  severity: string | null;
+  description: string | null;
+  reason: string | null;
```

Comment:
Hmm... I expected `string | undefined` rather than `string | null`. And what the API sends **is** `string | undefined`... But [our type](https://github.com/phylum-dev/cli/blob/3d9d4158da7a54d9d6900dab406e443be9f31077/cli/src/types.rs#L102-L112) doesn't use `#[skip_serializing_if = "Option::is_none"]`, which means that these fields will be given a value of `null` when serializing...

I think we should update our type to serialize the same way that API does. And then we can update these fields to use the `?` syntax for optional dependencies.

```suggestion
  tag?: string;
  domain?: string;
  severity?: string;
  description?: string;
  reason?: string;
```

Reply:
Yeah I much prefer the `?` syntax anyway. I just didn't use that because other feels seemed to use `| null` already.

Reply:
They are different. `foo: string | null` implies that `foo` will always exist in the object, but might be `null`.

On the other hand `foo?: string` (which is just fancy syntax for `foo: string | undefined`) means that `foo` may not exist in the object.